tree/ntuple/v7/inc/ROOT/libdaos_mock/daos.h

```diff
@@ -238,13 +243,80 @@ int daos_obj_update(daos_handle_t oh, daos_handle_t th, uint64_t flags,
 
 
 //////////////////////////////////////////////////////////////////////////////// daos_prop.h
-
+#include <ctype.h>
 
 /** daos properties, for pool or container */
 typedef struct {
 	char		unused; // silence [-Wextern-c-compat]
 } daos_prop_t;
 
+#define DAOS_PROP_LABEL_MAX_LEN (127)
+#define DAOS_PROP_MAX_LABEL_BUF_LEN (DAOS_PROP_LABEL_MAX_LEN + 1)
+
+/**
+ * Check if DAOS (pool or container property) label string is valid.
+ * DAOS labels must consist only of alphanumeric characters, colon ':',
+ * period '.', hyphen '-' or underscore '_', and must be of length
+ * [1 - DAOS_PROP_LABEL_MAX_LEN].
+ * \param[in] label	    Label string
+ * \return	  true      Label meets length/format requirements
+ *			  false     Label is not valid length or format
+ */
+static inline bool daos_label_is_valid(const char *label)
+{
+   int len;
+   int i;
+   bool maybe_uuid = false;
+
+   /** Label cannot be NULL */
+   if (label == NULL)
+      return false;
+
+   /** Check the length */
+   len = strnlen(label, DAOS_PROP_LABEL_MAX_LEN + 1);
+   if (len == 0 || len > DAOS_PROP_LABEL_MAX_LEN)
+      return false;
+
+   /** Verify that it contains only alphanumeric characters or :.-_ */
+   for (i = 0; i < len; i++) {
+      char c = label[i];
+
+      if (isalnum(c) || c == '.' || c == '_' || c == ':')
+         continue;
+      if (c == '-') {
+         maybe_uuid = true;
+         continue;
+      }
+
+      return false;
+   }
+
+   /** Check to see if it could be a valid UUID */
```

Comment:
Perhaps add a comment why we explicitly check for (and forbid) UUIDs

Reply:
This function is taken wholesale from a more recent version of Daos. The check explicitly forbids UUIDs while validating labels because the API so disambiguates IDs for pools or containers - e.g. `daos_cont_create` takes `uuid_t` and `daos_cont_create_with_label` takes `const char*` and generates a UUID internally. I presume the intention is to prevent the user from creating a container whose label matches the UUID of an existing container. 

Perhaps it is excessive to forbid UUID-like labels in our mock library, since it is now entirely label-based and does not hash them into `uuid_t` internally. 

For our `RDaos` backend layer, it would likely be sensible to support both forms of ID rather than switch to labels exclusively, like in this PR. In that case, I could switch back to representing pool and container IDs with `uuid_t` as long as the correct API functions are called at initialization and the hash is retrieved.